hack/presubmit.sh

```diff
@@ -67,15 +67,13 @@ update_codegen() {
 }
 
 combine() {
-  SOURCE_DIR=$1
-  REGEX=$2
-  DEST_YAML=$3
-
-  SOURCE_YAMLS=$(find ${SOURCE_DIR} -regextype sed -regex ${REGEX} | sort)
-  mkdir -p $(dirname $3)
-  cat "${SCRIPT_ROOT}"/hack/boilerplate.txt > $DEST_YAML
-  printf "\n# NOTE: This file is autogenerated.\n" >> $DEST_YAML
-  sed -s '$a---' $SOURCE_YAMLS | sed -e '$ d' -e '/^#/d' -e '/^$/d' >> $DEST_YAML
+  local source_dir=$1
+  local dest_yaml=$2
+
+  mkdir -p $(dirname $dest_yaml)
+  cat "${SCRIPT_ROOT}"/hack/boilerplate.txt > $dest_yaml
+  printf "\n# NOTE: This file is autogenerated.\n" >> $dest_yaml
+  kustomize build --reorder=none "${source_dir}" >> $dest_yaml
```

Comment:
Interesting, it seems kustomize is re-ordering the output despite specifying `--reorder=none`. I suppose this isn't the biggest deal since the generated manifests should only ever be read-only.

Reply:
I don't have a strong opinion, but running a final `kustomize cfg fmt` on the output files may get them back in order, but I think that can be optional.

Reply:
The reorder is for reording resources, not fields. For example, kustomize tries to put things like namespaces or webhooks first.

Reply:
Ahh got it. Makes sense.